server/src/main/java/org/apache/uniffle/server/ShuffleServerGrpcService.java

```diff
@@ -226,26 +227,21 @@ public void sendShuffleData(SendShuffleDataRequest req,
         return;
       }
       final long start = System.currentTimeMillis();
+      manager.markBufferInUse(requireBufferId);
```

Comment:
What will happen if manager.isPreAllocated -> clean require buffer -> manager.markBufferInUse

Reply:
The gc time of our server will be a little long. Will it increase the probability of this problem?

Reply:
> Well, this could happen, but in practice it should be extremely rare.

In current codebase, it is extremely rare too.


> Even if the above case happens, the outcome would be double releasing `usedMemory` size which has a safe bet that it will never decreased to less than zero.
It will lead to OOM

Reply:
> Well, this could happen, but in practice it should be extremely rare.

In current codebase, it is extremely rare too.


> Even if the above case happens, the outcome would be double releasing `usedMemory` size which has a safe bet that it will never decreased to less than zero.

It will lead to OOM

Reply:
Incorrect `preAllocatedSize` has no impact on the server, but Incorrect `usedMemory` has. The reason why i don't rush to repair it is because there may be a big adjustment after #133

Reply:
I mean `preAllocatedSize` should be removed.

Reply:
I mean should we modify the `usedMemory` problem in this pr by the way?